scipy/stats/_hypotests.py

```diff
@@ -132,6 +134,227 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
+    observed random variates X_1, ..., X_n that are assumed to be independent
+    and identically distributed ([1]_). The null hypothesis is that the X_i
+    have cumulative distribution F.
+
+    Parameters
+    ----------
+    rvs : array_like
+        The observed values of the random variables, it should be a 1-D array.
+    cdf : str or callable
+        If a string, it should be the name of a distribution in `scipy.stats`.
+        If a callable, that callable is used to calculate the cdf.
+        ``cdf(x, *args) -> float``, it should be a cdf.
+    args : tuple, optional
+        Distribution parameters. These are assumed to be known, see Notes.
+
+    Returns
+    -------
+    statistic : float
+        Cramer-von Mises statistic.
+    pvalue :  float
+        The p-value.
+
+    See Also
+    --------
+    kstest
+
+    Notes
+    -----
+    The p-value relies on the approximation given by equation 1.8 in [2]_.
+    It is important to keep in mind that the p-value is only accurate if
+    one tests a simple hypothesis, i.e. the parameters of the reference
+    distribution are known. If the parameters are estimated from the data
+    (composite hypothesis), the computed p-value is not reliable.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Cramer-von_Mises_criterion
+    .. [2] Csorgo, S. and Faraway, J. (1996). The Exact and Asymptotic
+           Distribution of Cramer-von Mises Statistics. Journal of the
+           Royal Statistical Society, pp. 221-234.
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    Test the assumption of normality. In this example, one accepts
+    the assumption of a normal distribution at the significance level
+    of alpha=0.05.
+
+    >>> np.random.seed(626)
+    >>> x = stats.norm.rvs(size=500)
+    >>> w, p = stats.cvm_test(x, 'norm')
+    >>> w, p
+    (0.06342154705518796, 0.792680516270629)
+
+    We can also use `args` to check whether the data is drawn from a
+    shifted normal distribution, assuming a shift of 2 when the
+    true shift is 2.1. In that case, the null hypothesis is rejected
+    at the significance level of alpha=0.05.
+
+    >>> y = x + 2.1
+    >>> w, p = stats.cvm_test(y, 'norm', args=(2,))
+    >>> w, p
+    (0.4798693195559657, 0.044782228803623814)
+
+    Using a callable argument for cdf, this is equivalent to the
+    following:
+
+    >>> cdf = stats.norm.cdf
+    >>> stats.cvm_test(y, cdf, args=(2,))
+
+    """
+    if isinstance(cdf, str):
+        cdf = getattr(distributions, cdf).cdf
+
+    vals = np.sort(np.asarray(rvs))
+
+    if vals.size <= 1:
+        raise ValueError('The sample must contain at least two observations.')
+    if vals.ndim > 1:
+        raise ValueError('The sample must be one-dimensional.')
+
+    n = len(vals)
+    cdfvals = cdf(vals, *args)
+
+    u = (2*np.arange(1, n+1) - 1)/(2*n)
+    w = 1/(12*n) + np.sum((u - cdfvals)**2)
+
+    if w > 4 and n >= 12:
```

Comment:
Could you note where this is from?

Reply:
Also test for the warning for full test coverage.

Reply:
I removed the condition, it can lead to inconsistent results, further comments below